workspace/read-manifest/src/index.ts

```diff
@@ -1,15 +1,56 @@
 import util from 'util'
 import { WORKSPACE_MANIFEST_FILENAME } from '@pnpm/constants'
-import { PnpmError } from '@pnpm/error'
 import path from 'node:path'
 import readYamlFile from 'read-yaml-file'
+import {
+  assertValidWorkspaceManifestCatalog,
+  assertValidWorkspaceManifestCatalogs,
+  checkDefaultCatalogIsDefinedOnce,
+  type WorkspaceCatalog,
+  type WorkspaceNamedCatalogs,
+} from './catalogs'
+import { InvalidWorkspaceManifestError } from './errors/InvalidWorkspaceManifestError'
 
 export interface WorkspaceManifest {
   packages?: string[]
+
+  /**
+   * The default catalog. Package manifests may refer to dependencies in this
+   * definition through the `catalog:default` specifier or the `catalog:`
+   * shorthand.
+   */
+  catalog?: WorkspaceCatalog
+
+  /**
+   * A dictionary of named catalogs. Package manifests may refer to dependencies
+   * in this definition through the `catalog:<name>` specifier.
+   */
+  catalogs?: WorkspaceNamedCatalogs
+}
+
+export interface ReadWorkspaceManifestOptions {
+  /**
+   * Whether or not to read the catalog and catalogs fields. This is currently
+   * disabled by default while the overall catalogs feature is in development.
+   *
+   * @default false
+   */
+  readonly catalogs?: boolean
 }
 
-export async function readWorkspaceManifest (dir: string): Promise<WorkspaceManifest | undefined> {
+export async function readWorkspaceManifest (
+  dir: string,
+  opts?: ReadWorkspaceManifestOptions
+): Promise<WorkspaceManifest | undefined> {
   const manifest = await readManifestRaw(dir)
+
+  // Disable catalogs config reads by default until the overall feature is ready.
+  const isCatalogsConfigEnabled = opts?.catalogs ?? false
+  if (!isCatalogsConfigEnabled && typeof manifest === 'object' && manifest != null) {
+    delete (manifest as { catalog?: unknown }).catalog
+    delete (manifest as { catalogs?: unknown }).catalogs
+  }
```

Comment:
The intention of feature flagging this was to prevent validation errors if users attempted to define catalogs today. I think a user would reasonably assume catalogs were ready as an overall feature if pnpm validated its config, but that's not true today.